### PR TITLE
[feature] simple pre-flight auth checks

### PR DESCRIPTION
**Pull request by ryanking**
Before running any terraform commands, run a quick authentication check.

This won't catch every problem (like authorization failures), but in the cases where a profile is not configured at all, this should give a better error message than terraform does.

Thanks to @kislyuk  for pushing for a better experience here.

Note that his assumes the aws cli is installed. Do we think that is a fair assumption?

### Test Plan
* make test
* running a prerelease on shared-infra

### References
(Optional) Additional links to provide more context.

### CODE REVIEW

File: testdata/bless_provider/scripts/component.mk [hunk 1]
```diff
@@ -42,11 +42,16 @@ lint-terraform-fmt: terraform
 	@echo
 .PHONY: lint-terraform-fmt
 
+check-auth:
+	@aws --profile $(AWS_BACKEND_PROFILE) sts get-caller-identity >/dev/null || echo "AWS AUTH error. This component is configured to use a profile name $(AWS_BACKEND_PROFILE). Please add one to your ~/.aws/config"
+	@aws --profile $(AWS_PROVIDER_PROFILE) sts get-caller-identity >/dev/null || echo "AWS AUTH error. This component is configured to use a profile name $(AWS_PROVIDER_PROFILE). Please add one to your ~/.aws/config"
```
Comment (edulop91) [inline testdata/bless_provider/scripts/component.mk]: what if we have components without an aws provider configured?

Comment (ryanking) [reply testdata/bless_provider/scripts/component.mk]: This should handle it https://github.com/chanzuckerberg/fogg/pull/366/files#diff-d1dfe7c391918134f1692e607df2c103R9 ?

Comment (ryanking) [reply testdata/bless_provider/scripts/component.mk]: I will test on snowflake-infra

Comment (ryanking) [reply testdata/bless_provider/scripts/component.mk]: derp nevermind, still won't work

Comment (ryanking) [reply testdata/bless_provider/scripts/component.mk]: fixed

File: templates/repo/scripts/component.mk [hunk 2]
```diff
@@ -42,11 +42,17 @@ lint-terraform-fmt: terraform
 	@echo
 .PHONY: lint-terraform-fmt
 
+check-auth:
+	@for p in $(AWS_BACKEND_PROFILE) $(AWS_PROVIDER_PROFILE); do \
```
Comment (edulop91) [inline templates/repo/scripts/component.mk]: :+1: nice

**Pull request merged**